### PR TITLE
feat: add API for DataPlane managed PDB

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

Introduce API for DataPlane managed PDB.

**Which issue this PR fixes**

Part of #142.

**Special notes for your reviewer**:

`PodDisruptionBudgetSpec` has been copied from https://pkg.go.dev/k8s.io/api@v0.30.3/policy/v1#PodDisruptionBudgetSpec with `Selector` removed as that is going to be managed by the operator to match `DataPlane`'s Pods.

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect significant changes

### CODE REVIEW

File: api/v1beta1/dataplane_types.go [hunk 1]
```diff
@@ -64,6 +65,75 @@ type DataPlaneOptions struct {
...
+	PodDisruptionBudget *PodDisruptionBudget `json:"podDisruptionBudget,omitempty"`
+}
+
+// PodDisruptionBudget defines the configuration for the PodDisruptionBudget.
+type PodDisruptionBudget struct {
+	// Enabled indicates whether the PodDisruptionBudget should be created.
+	//
+	// +optional
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled"`
```
Comment (czeslavo) [inline api/v1beta1/dataplane_types.go]: @pmalek Why do we need this flag? As `resources.podDisruptionBudget` is optional, cannot we support enabling it simply by making it non-nil?

Comment (pmalek) [reply api/v1beta1/dataplane_types.go]: My reasoning for this was to allow users to keep the spec but disable the pdb (for testing purposes or to temporarily suspend).

That use case might be contrived though.

Comment (pmalek) [reply api/v1beta1/dataplane_types.go]: On a side note without the `enabled` field how do we consider the pdb enabled? If any of the spec fields is non nil? Or do we change the `spec` to be a pointer and check whether that's non nil?

Comment (czeslavo) [reply api/v1beta1/dataplane_types.go]: > My reasoning for this was to allow users to keep the spec but disable the pdb (for testing purposes or to temporarily suspend).

I'd drop it for simplicity and add only if there's a specific request for that. 

> On a side note without the enabled field how do we consider the pdb enabled? If any of the spec fields is non nil? Or do we change the spec to be a pointer and check whether that's non nil?

`DataPlaneResources.PodDisruptionBudget` is a pointer that can be made explicitly optional. If it is not `nil`, that means we should create a PDB based on the spec. Do I miss something?

### PDB disabled (`resources.podDistruptionBudget` pointer is nil)
```yaml
apiVersion: gateway-operator.konghq.com/v1beta1
kind: DataPlane
metadata:
  name: dataplane-with-pdb
spec:
  # Do not fill resources.podDisruptionBudget to not create a PDB for DP.
  # resources:
  #   podDisruptionBudget:
  deployment:
    replicas: 3
```

### PDB enabled with an empty spec (apparently it is allowed, I'm not sure what it effectively means if there's no minAvailable/maxUnavailable set)
```yaml
apiVersion: gateway-operator.konghq.com/v1beta1
kind: DataPlane
metadata:
  name: dataplane-with-pdb
spec:
  resources:
    podDisruptionBudget: {}
  deployment:
    replicas: 3
```

Comment (pmalek) [reply api/v1beta1/dataplane_types.go]: 👍 

Removed `enabled` field and added CEL validation as per the docs: https://kubernetes.io/docs/tasks/run-application/configure-pdb/#specifying-a-poddisruptionbudget

> You can specify only one of maxUnavailable and minAvailable in a single PodDisruptionBudget.

**Review by czeslavo — APPROVED** [review]

**Pull request merged**